lib/jekyll/filters.rb

```diff
@@ -373,8 +373,8 @@ def time(input)
       when Numeric
         Time.at(input)
       else
-        Jekyll.logger.error "Invalid Date:", "'#{input}' is not a valid datetime."
-        exit(1)
+        raise Errors::InvalidDateError,
+          "Invalid Date: '#{input}' is not a valid datetime."
```

Comment:
I'm not sure about error message here.
Theoretically it should only get here if input is not given, so maybe it should be something like:
`No input given to filter #{filter_name} in #{document_path}` ?
Can we determine somehow the path of the document where the error is?